### PR TITLE
[optimizer] Adam

### PR DESCRIPTION
**Pull request by RodionovF**

### CODE REVIEW

**Comment by RodionovF** [comment]
output Sample MNIST with ADAM

```
Epoch: 0
Loss: 0.786745
Loss: 0.559453
Loss: 0.473762
Loss: 0.414654
Loss: 0.370882
Loss: 0.343476
Loss: 0.321217
Loss: 0.3048
Loss: 0.289838
Loss: 0.276903
Loss: 0.264504
Loss: 0.255388
Loss: 0.246734
Loss: 0.237692
Loss: 0.230381
Loss: 0.222062
Loss: 0.216012
Loss: 0.210831
Loss: 0.207552
Accuracy: 0.962039
```

File: include/deepworks/optimizer.hpp [hunk 1]
```diff
@@ -34,9 +35,35 @@ class SGDMomentum {
...
+
+    void step();
+
+    float get_lr() const;
+
+    void set_lr(float lr);
+
+private:
+    float m_lr;
+    float m_beta_one;
```
Comment (TolyaTalamanov) [inline include/deepworks/optimizer.hpp:57]: https://pytorch.org/docs/stable/optim.html#torch.optim.Adam

According to pytorch it looks like a `std::array<float, 2>` as option. Honestly speaking I don't know, up to you

**Review by TolyaTalamanov — APPROVED** [review]
Guys have a look, please @l-bat 

LGTM :+1:  :fire:

**Comment by TolyaTalamanov** [comment]
Thanks! 🔥

**Pull request merged**